js/src/helpers/addCheckToChecklist.js

```diff
@@ -0,0 +1,93 @@
+import { __ } from "@wordpress/i18n";
+import getIndicatorForScore from "../analysis/getIndicatorForScore";
+
+/**
+ * Adds a "No focus keyword was entered" result when no focus keyphrase
+ * was entered.
+ *
+ * @param {Object[]} checklist The checklist on which to add the check.
+ * @param {Object}   store     The Yoast SEO redux store.
+ *
+ * @returns {void}
+ */
+export function maybeAddFocusKeyphraseCheck( checklist, store ) {
+	const focusKeyphrase = store.getFocusKeyphrase();
+
+	if ( ! focusKeyphrase ) {
+		checklist.push( {
+			label: __( "No focus keyword was entered", "wordpress-seo" ),
+			score: "bad",
+		} );
+	}
+}
+
+/**
+ * Adds the readability check if readability analysis is active.
+ *
+ * @param {Object[]} checklist The checklist on which to add the readability check.
+ * @param {Object}   store     The Yoast SEO redux store.
+ *
+ * @returns {void}
+ */
+export function maybeAddReadabilityCheck( checklist, store ) {
+	const { isKeywordAnalysisActive } = store.getPreferences();
+
+	if ( isKeywordAnalysisActive ) {
+		const readabilityScoreIndicator = getIndicatorForScore( store.getReadabilityResults().overallScore );
+
+		checklist.push( {
+			label: __( "Readability analysis:", "wordpress-seo" ),
+			score: readabilityScoreIndicator.className,
+			scoreValue: readabilityScoreIndicator.screenReaderReadabilityText,
+		} );
+	}
+}
+
+/**
+ * Adds the SEO check if SEO analysis is active.
+ *
+ * @param {Object[]} checklist The checklist on which to add the SEO score.
+ * @param {Object}   store     The Yoast SEO redux store.
+ *
+ * @returns {void}
+ */
+export function maybeAddSEOCheck( checklist, store ) {
+	const { isContentAnalysisActive } = store.getPreferences();
+
+	if ( isContentAnalysisActive ) {
+		const seoScoreIndicator = getIndicatorForScore( store.getResultsForFocusKeyword().overallScore );
+
+		checklist.push( {
+			label: __( "SEO analysis:", "wordpress-seo" ),
+			score: seoScoreIndicator.className,
+			scoreValue: seoScoreIndicator.screenReaderReadabilityText,
+		} );
+	}
+}
+
+/**
+ * Adds a score item for the schema blocks validation results, indicating whether all the
+ * schema blocks on the page have valid schema.
+ *
+ * This check is hidden when no schema validation results are known
+ * (e.g. when no schema blocks exist on the page).
+ *
+ * @param {Object[]} checklist The score items to add the score to.
+ * @param {Object}   store     The Yoast SEO redux store.
+ *
+ * @returns {void}
+ */
+export function maybeAddSchemaBlocksValidationCheck( checklist, store ) {
+	const schemaBlocksValidationResults = store.getSchemaBlocksValidationResults();
+	const validationResults = Object.values( schemaBlocksValidationResults );
```

Comment:
```suggestion
	const validationResults = Object.values( schemaBlocksValidationResults.validations );
```

Reply:
I decided [to return it directly in the selector instead](https://github.com/Yoast/wordpress-seo/pull/16598/commits/e2ccc40a52b582236fe0c944d0b2475f1d71da15#diff-7d3913dffe6b9aebead0eafd97a64002ac9a7b1d244f057e26803d4e32a0f0deR9).